bot/cogs/doc.py

```diff
@@ -0,0 +1,207 @@
+        """
+        Given a Python symbol, return its signature and description.
+
+        :param symbol: The symbol for which HTML data should be returned.
+        :return:
+        A tuple in the form (str, str), or `None`.
+        The first tuple element is the signature of the given
+        symbol as a markup-free string, and the second tuple
+        element is the description of the given symbol with HTML
+        markup included. If the given symbol could not be found,
+        returns `None`.
+        """
+
+        url = self.inventories.get(symbol)
+        if url is None:
+            return None
+
+        async with aiohttp.ClientSession() as cs:
+            async with cs.get(url) as response:
+                html = await response.text(encoding='utf-8')
+
+        # Find the signature header and parse the relevant parts.
+        symbol_id = url.split('#')[-1]
+        soup = BeautifulSoup(html, 'html.parser')
+        symbol_heading = soup.find(id=symbol_id)
+        signature_buffer = []
+
+        for tag in symbol_heading.strings:
+            if tag not in UNWANTED_SIGNATURE_SYMBOLS:
+                signature_buffer.append(tag.replace('\\', ''))
+
+        signature = ''.join(signature_buffer)
+        description = str(symbol_heading.next_sibling.next_sibling).replace('¶', '')
+
+        return signature, description
+
+    @async_cache(arg_offset=1)
+    async def get_symbol_embed(self, symbol: str) -> Optional[discord.Embed]:
+        """
+        Using `get_symbol_html`, attempt to scrape and
+        fetch the data for the given `symbol`, and build
+        a formatted embed out of its contents.
+
+        :param symbol: The symbol for which the embed should be returned
+        :return:
+        If the symbol is known, an Embed with documentation about it.
+        Otherwise, `None`.
+        """
+
+        scraped_html = await self.get_symbol_html(symbol)
+        if scraped_html is None:
+            return None
+
+        signature = scraped_html[0]
+        permalink = self.inventories[symbol]
+        description = markdownify(scraped_html[1])
+        if len(description) > 1000:
+            description = description[:1000] + f"... [read more]({permalink})"
+
+        description = WHITESPACE_AFTER_NEWLINES_RE.sub('', description)
+
+        if not signature:
+            # It's some "meta-page", for example:
+            # https://docs.djangoproject.com/en/dev/ref/views/#module-django.views
+            return discord.Embed(
+                title=f'`{symbol}`',
+                url=permalink,
+                description="This appears to be a generic page not tied to a specific symbol."
+            )
+
+        return discord.Embed(
+            title=f'`{symbol}`',
+            url=permalink,
+            description=f"```py\n{signature}```{description}"
+        )
+
+    @commands.command(name='doc()', aliases=['doc'])
+    async def doc(self, ctx, *, symbol: commands.clean_content):
```

Comment:
If not found, perform a fuzzy search (fuzzywuzzy or faster implementation which I can give you)

Reply:
`fuzzywuzzy` is licensed under GPL2... doesn't that mean we'd have to use GPL2 for this repo, or did I misunderstand how GPL2 works?

Reply:
MIT I guess, had this in my gists somewhere

Reply:
we can use fuzzywuzzy.

Reply:
fuzzywuzzy is slower though, but sure. I'm not sure about licensing, but my code is an option. If fuzzywuzzy's performance turns out to be fast enough, we can definitely use it.